### ISSUE TITLE
distanceTo between Polygons: First point of segment is not always on the 'this' polygon

### ISSUE THREAD

**Issue by gigerdo** [alexbol99/flatten-js#57]
Version: 1.2.10

According to documentation this should be the case. An example to show the problem:

```
import Flatten from "@flatten-js/core";
import Box = Flatten.Box;
import Polygon = Flatten.Polygon;

function buildPoly(x: number, y: number) {
    let width = 200;
    let height = 50;
    let rec = new Box(
        x,
        y,
        x + width,
        y + height
    );
    return new Polygon(rec);
}

const p1 = buildPoly(0, 0);
const p2 = buildPoly(0, 500);

const [distance, segment] = p1.distanceTo(p2);

console.log("Polygons", p1.vertices, p2.vertices)
console.log("Segment Start: ", segment.start, p1.contains(segment.start));
console.log("Segment End: ", segment.end, p2.contains(segment.end));
```
Which results in segment.start being part of p2, and not p1 as expected.

**Comment by alexbol99** [comment]
Hi @gigerdo ,

Bug fixed in the patch v1.2.12

Thank you for contribution to this library,
Alex Bol

**Comment by gigerdo** [comment]
Thanks for the fast fix. I love the library so far, it is really well designed and easy to use.
